code/modules/projectiles/gun_system.dm

```diff
@@ -114,6 +114,19 @@
 
 	var/general_codex_key = "guns"
 
+	///The mob holding the gun
+	var/mob/gun_user
+	///The atom targeted by the user
+	var/atom/target
```

Comment:
Similarly on the other comment, this is stuff that should be handled by the component, not by the gun.

Reply:
the aiming part should be handled by another component, an aiming component rather than a shooting one, both are pretty different

Reply:
Thats sounds like a terrible idea splinter but yes this should be on the component